client/lib/analytics/ad-tracking.js

```diff
@@ -994,7 +1032,18 @@ function recordOrderInFacebook( cart, orderId ) {
 		return;
 	}
 
-	debug( 'recordOrderInFacebook: Record purchase' );
+	const containsJetpackPlan = some( cart.products, productsValues.isJetpackPlan );
+	const containsNonJetpackProduct = some( cart.products, p => {
+		return ! productsValues.isJetpackPlan( p );
+	} );
+
+	if ( containsJetpackPlan && containsNonJetpackProduct ) {
```

Comment:
That seems like it could work. I actually was unable to test this scenario because it doesn't appear possible at this time; i.e., I couldn't find a way to get Jetpack + something at WP.com into my cart at the same time.